### PR TITLE
Add protocolfilter setting to set overriding protocols for an origin.

### PR DESCRIPTION
**Pull request by ghost**
b/205134049

### CODE REVIEW

**Review by jellefoks — APPROVED** [review]
Awesome!

File: cobalt/h5vcc/h5vcc_settings.cc [hunk 1]
```diff
@@ -118,6 +118,15 @@ bool H5vccSettings::Set(const std::string& name, SetValueType value) const {
     }
   }
 
+  if (name.compare(network::kProtocolFilterKey) == 0 &&
+      value.IsType<std::string>() && value.AsType<std::string>().size() < 512) {
```
Comment (jellefoks) [inline cobalt/h5vcc/h5vcc_settings.cc:122]: Oh, I think a max length of 512 is too short. Let's set it nice and high, like 16384 or so.

**Comment by jellefoks** [comment]
Nice. With a manual test this last version works as expected.

File: cobalt/h5vcc/h5vcc_settings.cc [hunk 2]
```diff
@@ -118,6 +118,31 @@ bool H5vccSettings::Set(const std::string& name, SetValueType value) const {
...
+  if (name.compare(network::kProtocolFilterKey) == 0 &&
+      value.IsType<std::string>() &&
+      value.AsType<std::string>().size() < 16384) {
+    std::string raw_json = value.AsType<std::string>();
+    base::Value old_config_json;
+    persistent_settings_->Get(network::kProtocolFilterKey, &old_config_json);
+
+    if (raw_json.empty() && (!old_config_json.is_string() ||
+                             !old_config_json.GetString().empty())) {
+      persistent_settings_->Set(network::kProtocolFilterKey, base::Value());
```
Comment (jellefoks) [inline cobalt/h5vcc/h5vcc_settings.cc:130]: This appears to be missing a `network_module_->SetProtocolFilterUpdatePending();`

**Review by jellefoks — APPROVED** [review]
Nice! Just one comment for an early return.

**Review by kaidokert — APPROVED** [review]

**Pull request merged**